mne/preprocessing/flat.py

```diff
@@ -0,0 +1,99 @@
+# Authors: Eric Larson <larson.eric.d@gmail.com>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+
+from ..io import BaseRaw
+from ..io.pick import _picks_to_idx
+from ..utils import (_validate_type, verbose, logger, _pl,
+                     _mask_to_onsets_offsets, ProgressBar)
+
+
+@verbose
+def mark_flat(raw, bad_percent=5., min_duration=0.005, picks=None,
+              verbose=None):
+    r"""Mark flat segments of raw data using annotations or in info['bads'].
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data.
+    bad_percent : float
+        The percentage of the time a channel can be bad.
+        Below this percentage, temporal bad marking (:class:`~mne.Annotations`)
+        will be used. Above this percentage, spatial bad marking
+        (:class:`info['bads'] <mne.Info>`) will be used.
+        Defaults to 5 (5%%).
+    min_duration : float
+        The minimum duration (sec) to consider as actually flat.
+        For some systems with low bit data representations, adjacent
+        channels with exactly the same value are not totally uncommon.
```

Comment:
that should read "adjacent samples", right?

Reply:
Yep, fixed in #6103